## Leading Causes of Death United States of America
## Introduction
In this report, we will be discussing the leading causes of mortality in the United States. Through our analysis we will be generating tables to determine the annual change in mortality in the United States over seven years (2010-2016), we will also find the top four leading causes of death both at the National and State level. Furthermore, we will also ascertain if the leading causes at both levels hold some similarities. Finally, we will determine if there are changes in the four leading causes of death annually nationwide.
### Data
For this project, we will be analyzing publicly available data from the United States National Center for Health and United States Census Bureau. For sake of our analysis, we have filtered the years in both datasets from 2010-2016.

## Data Analysis

We will be conducting an exploratory analysis of the datasets to gain insights using the Python package **Pandas**. Our analysis will involve:
1. The preliminary analysis of the datasets
2. Cleaning, filtering and merging the datasets
3. Conducting analytical calculations
4. Deriving Insights

### Preliminary Analysis
The first step of our analysis is to import the Pandas library and conduct a preliminary analysis to ascertain what sort of datasets we have, what exactly is inside them, and determine whether they require data cleaning. To accomplish this we use the functions and methods like **read_csv()**, **read_excel()**, **head()**, **.shape** and **columns()**.

In [1]:
#Loading the Libraries
import pandas as pd
#Reading the data from a csv and excel file
data = pd.read_excel(r"C:\Users\mahno\OneDrive\Desktop\MS Business Analytics\Fall Semester\BANA 680\Assignments\Assignment 2\nst-est2018-01.xlsx", header=None)
data1 = pd.read_csv(r'C:\Users\mahno\OneDrive\Desktop\MS Business Analytics\Fall Semester\BANA 680\Assignments\Assignment 2\NCHS_-_Leading_Causes_of_Death__United_States.csv')

In [ ]:
#Number of Rows and Columns
print("Dataset 1 number of rows and columns:" ,data.shape)
print("Dataset 2 number of rows and columns:" ,data1.shape)

In [ ]:
#Column Names in datasets
print("Dataset 1 column names:" ,data.columns)
print("Dataset 2 column names:" ,data1.columns)

In [ ]:
#Top 3 rows of dataset
data.head(n=3)

In [ ]:
#Top 3 rows of dataset
data1.head(n=3)

By conducting our preliminary analysis we discover the scope of our datasets. Dataset 1 contains the population of the United States as a whole, regions, and at the state level from 2010 to 2018. While dataset 2 contains the number of deaths by various causes at the State level from 1999 to 2016.

### Data Cleaning

Through our preliminary analysis, we notice that the datasets have some redundant data in the form of NaN values and extra columns and rows that are irrelevant to our analysis. In addition to that, we also observe that the structure and shape of both datasets are not compatible with each other. As such it is necessary to clean the data and organize them to make them compatible with our calculations.

For this we filter out relevant rows and columns using the **.loc** attribute and **drop()** function. We then restructure the data using the **transpose()** function and drop decimals from numeric values using **astype()** function. We rename the columns to the correct format using the **rename()** and **columns()** functions. After the data has been thoroughly cleaned we convert it from wide to long-form using the **melt()** function and merge them using the **merge()** function.

In [2]:
#Filtering out data
data = data.loc[3:59,:]
#Dropping irelevant rows
data = data.drop([5,6,7,8])
#Dropping irelevant columns
data = data.drop(columns=data.columns[[1,2,10,11]])
#Resetting index for better understanding and aesthetics
data.reset_index(drop=True, inplace=True)
#Tansposing the data
data = data.transpose()
#Renaming the columns
data = data.rename(columns=data.iloc[0,:])
data = data.drop([0])
#Resetting index for better understanding and aesthetics
data.reset_index(drop=True, inplace=True)
#Dropping decimals
data = data.astype(int)
#Viewing data
data.head(n=3)

NaN  United States  .Alabama  .Alaska  .Arizona  .Arkansas  .California  \
0  2010      309326085   4785448   713906   6407774    2921978     37320903   
1  2011      311580009   4798834   722038   6473497    2940407     37641823   
2  2012      313874218   4815564   730399   6556629    2952109     37960782   

   .Colorado  .Connecticut  .Delaware  ...  .South Dakota  .Tennessee  \
0    5048281       3579125     899595  ...         816165     6355301   
1    5121771       3588023     907316  ...         823484     6397410   
2    5193721       3594395     915188  ...         833496     6451281   

     .Texas    .Utah  .Vermont  .Virginia  .Washington  .West Virginia  \
0  25242679  2775334    625880    8023680      6742902         1854214   
1  25646227  2814216    626979    8100469      6821655         1856074   
2  26089620  2853467    626063    8185229      6892876         1856764   

   .Wisconsin  .Wyoming  
0     5690479    564483  
1     5704755    567224  
2     5719855    576270  

[3 rows x 53 columns]

In [3]:
#Renaming columns to remove special characters
data.columns = ['Year',"United States",'Alabama','Alaska','Arizona','Arkansas',"California",'Colorado','Connecticut','Delaware',"District of Columbia",'Florida','Georgia','Hawaii','Idaho','Illinois','Indiana','Iowa','Kansas','Kentucky','Louisiana','Maine','Maryland','Massachusetts','Michigan','Minnesota','Mississippi','Missouri','Montana','Nebraska','Nevada','New Hampshire','New Jersey','New Mexico','New York','North Carolina','North Dakota','Ohio','Oklahoma','Oregon','Pennsylvania','Rhode Island','South Carolina','South Dakota','Tennessee','Texas','Utah','Vermont','Virginia','Washington','West Virginia','Wisconsin','Wyoming']

In [4]:
#Converting to Long form 
data=pd.melt(data,id_vars=['Year'],var_name='State', value_name='Population')

In [5]:
data = pd.merge(data1, data, on=["Year","State"])

In [6]:
#Removing Irrelevant columns from Dataset
data = data.drop(['113 Cause Name', 'Age-adjusted Death Rate'], axis = 1)
#Final Cleaned, Filtered and Merged dataset
data

Year               Cause Name    State  Deaths  Population
0     2011                  Suicide  Alabama     654     4798834
1     2011  Influenza and pneumonia  Alabama     951     4798834
2     2011           Kidney disease  Alabama    1047     4798834
3     2011                 Diabetes  Alabama    1278     4798834
4     2011      Alzheimer's disease  Alabama    1486     4798834
...    ...                      ...      ...     ...         ...
3999  2011   Unintentional injuries  Wyoming     317      567224
4000  2011                     CLRD  Wyoming     327      567224
4001  2011                   Cancer  Wyoming     936      567224
4002  2011            Heart disease  Wyoming     926      567224
4003  2011               All causes  Wyoming    4387      567224

[4004 rows x 5 columns]

### United States Annual Changes in Mortality rate

#### Analysis

To determine if Americans are facing increasing, decreasing, or steady likelihood of death, we filter out **All causes** from the Cause Name column and **United States** from the State column so that the data doesn't repeat itself in our calculations. We then group our dataset by **Year** and sum the number of deaths in each year. This gives us the total number of deaths in the United States each year. Moreover, we standardize the results by dividing them by the total population of that year and getting a percentage. Our final result is a table that depicts the National Annual Change in Mortality rate.

In [7]:
data_year = data
#Removing data so it doesn't repeat itself in the table
data_year = data_year[data_year["Cause Name"] != "All causes"]
data_year = data_year[data_year["State"] != "United States"]
#Total Number of Deaths grouped by Year
dyear= data_year.groupby(["Year"]).sum()
dyear["Death Percentage"] = (dyear["Deaths"]/(dyear["Population"]/10)*100)
del dyear['Population']
dyear

Deaths  Death Percentage
Year                           
2010  1852349          0.598834
2011  1869321          0.599949
2012  1876588          0.597879
2013  1910311          0.604418
2014  1938408          0.608822
2015  2013017          0.627611
2016  2034119          0.629619

#### Insights

According to our analysis, we conclude that the U.S death rate is slowly increasing. We analyzed the data from the year 2010 to the year 2016. The total number of deaths increased from 1,852,349 in 2010 to 2,034,119 in 2016; meanwhile, the death percentage increased from 0.59% in 2010 to 0.63% in 2016. Thus, we come to our conclusion that Americans are facing an increased likelihood of death. 

### Four Leading Causes of Death Nationwide and Year by Year changes

#### Analysis

For finding out the four leading causes of death for Americans and how they are changing annually, we remove **All causes** from the Cause Name column and only keep **United States** in the State column. This removes all the irrelevant data for our analysis. We then group our dataset by **State**, **Year**, and  **Cause Name** and sum the number of deaths in each year. This gives us the total number of deaths by each disease in the United States. We sort the data from largest to smallest and filter out the top four causes year-wise. Finally, we standardize the results by dividing them by the total population of that year and getting a percentage. Our final result is a table that depicts the top four leading causes of death in the U.S from 2010 to 2016.

In [10]:
#Creating a new data frame for Nationwide data
dataset1 = data
#Removing the irrelevant rows for the particular question
dataset1 = dataset1[dataset1["Cause Name"] != "All causes"]
dataset1 = dataset1[dataset1["State"] == "United States"]
#Grouping by State Year and Cause and getting sum of all categories of the causes
dataset1.groupby(['State',"Year",'Cause Name']).sum()
#Sorting values from largest to smallest
dataset1 = dataset1.sort_values(by = ["State","Year","Deaths"], ascending = False).groupby('Year').head(4)
dataset1.reset_index(drop=True, inplace=True)
#Calculating 
dataset1["Death Percentage"] = (dataset1["Deaths"]/dataset1["Population"])*100
dataset1.head(n=11)

Year              Cause Name          State  Deaths  Population  \
0   2016           Heart disease  United States  635260   323071342   
1   2016                  Cancer  United States  598038   323071342   
2   2016  Unintentional injuries  United States  161374   323071342   
3   2016                    CLRD  United States  154596   323071342   
4   2015           Heart disease  United States  633842   320742673   
5   2015                  Cancer  United States  595930   320742673   
6   2015                    CLRD  United States  155041   320742673   
7   2015  Unintentional injuries  United States  146571   320742673   
8   2014           Heart disease  United States  614348   318386421   
9   2014                  Cancer  United States  591700   318386421   
10  2014                    CLRD  United States  147101   318386421   

    Death Percentage  
0           0.196631  
1           0.185110  
2           0.049950  
3           0.047852  
4           0.197617  
5           0.185797  
6           0.048338  
7           0.045697  
8           0.192957  
9           0.185843  
10          0.046202

#### Insights

From our table, we can observe that the top four leading causes at the national level for the most part are the same. For our range of 2010 to 2016 **Heart disease**, **Cancer** and **CLRD** remain the top three, but from 2010 to 2012 **Stroke** was the fourth leading cause while from 2013 to 2016 **Unintentional injuries** was the fourth cause of death for Americans. As for the number of deaths by each cause we conclude that they show an upward trend for the most part. There are some years in which the number of deaths for a particular disease decreases.

### Four Leading Causes of Death Statewise and Year by Year changes

#### Analysis

To ascertain the four leading causes of death for Americans at the State level and how they are changing annually, we remove **All causes** from the Cause Name column and remove **United States** in the State column. This removes all the data at the national level and keeps only the State level data. After that, we group our dataset by **State**, **Year**, and  **Cause Name** and sum the number of deaths in each year. This gives us the total number of deaths by each disease at the State level. We sort the data from largest to smallest and filter out the top four causes year-wise. Finally, we standardize the results by dividing them by the total population of that year and getting a percentage. Our final result is a table that depicts the top four leading causes of deaths in the U.S at the State level from 2010 to 2016.

In [9]:
#Creating a new data frame for Nationwide data
dataset2 = data
#Removing the irrelevant rows for the particular question
dataset2 = dataset2[dataset2["Cause Name"] !="All causes"]
dataset2 = dataset2[dataset2["State"] !="United States"]
#Grouping by State Year and Cause
dataset2.groupby(['State',"Year",'Cause Name']).sum()
#Sorting values from largest to smallest
dataset2 = dataset2.sort_values(by = ["State","Year","Deaths"], ascending = False).groupby(['State','Year']).head(4)
dataset2.reset_index(drop=True, inplace=True)
dataset2["Death Percentage"] = (dataset2["Deaths"]/dataset2["Population"])*100
#Generating output for review
dataset2

Year              Cause Name    State  Deaths  Population  \
0     2016           Heart disease  Wyoming    1051      584290   
1     2016                  Cancer  Wyoming     962      584290   
2     2016  Unintentional injuries  Wyoming     371      584290   
3     2016                    CLRD  Wyoming     327      584290   
4     2015           Heart disease  Wyoming    1030      585668   
...    ...                     ...      ...     ...         ...   
1423  2011  Unintentional injuries  Alabama    2662     4798834   
1424  2010           Heart disease  Alabama   12083     4785448   
1425  2010                  Cancer  Alabama   10196     4785448   
1426  2010                    CLRD  Alabama    2866     4785448   
1427  2010                  Stroke  Alabama    2619     4785448   

      Death Percentage  
0             0.179876  
1             0.164644  
2             0.063496  
3             0.055965  
4             0.175868  
...                ...  
1423          0.055472  
1424          0.252495  
1425          0.213063  
1426          0.059890  
1427          0.054728  

[1428 rows x 6 columns]

#### Insights

Looking at the table for causes of deaths at the State level we observe that almost all of the States show **Heart disease** and **Cancer** in their top four causes. There are some States that have the same top four causes as that of the National ones but not all of them share this trend. The only similarity that we can observe is that **Heart disease** and **Cancer** are in the top four at both the National level and State level.

### Conclusion
By studying the tables of our analysis we can conclude that Americans are facing an increased likelihood of mortality as the years' progress, despite the increase in the population. The top four causes of death for Americans at the National level are:
1. Heart Disease
2. Cancer
3. CRLD
4. Stroke or Unintentional Injuries

Comparing the tables at the State level and National level we conclude that despite some States sharing the National trend the top four causes for all States are not the same. This may be because of the changes in geography, gender, ethnicity and economic state of both the individual and the area. The only similarity that States share with the National trend is that they have **Heart Disease** and **Cancer** in their top four causes of death.

#### Future Refinements
In our analysis, we took data from the years 2010 to 2016 due to the paucity of data. As such the conclusions drawn from the current dataset may not reflect the present trends. In the present, causes of death for Americans differ from past trends due to a variety of reasons. Some of them being:
1. Increasing obesity
2. Climate change
3. Social and Economic privation
4. Lack of affordable healthcare
5. New Super viruses

As such it would be amiss not to mention that scarcity of data may have lead us to these results. For future analysis, the use of a more complete dataset may lead to different and accurate results.